### PR TITLE
Windows: Produce a static version of the public libraries, always

### LINKED ISSUE #7492: Fix internal tests for shared Windows builds
The testsuite has many so called _internal_ tests, which link statically against libcrypto and/or libssl. All these internal tests are disabled for shared Windows builds, because the names of the static libraries collide with the names of the import libraries for the DLLs.

https://github.com/openssl/openssl/blob/f81b043ad856d8b9af5239a4978f8bd4b965dab9/test/build.info#L434-L445

There has been some discussion about whether the static libraries should be renamed, but that is only an option for the next major release. I would like to suggest a temporary workaround which would be backportable to 1.1.1 and would enable us to activate all internal tests for shared Windows builds.

Instead of renaming the _installed_ libraries, one could build additional static libraries (on windows only) to make the static tests depending on these. One does not even have to rename the static libraries, just place them in a separate directory (e.g. `static\libcrypto.lib`, `static\libssl.lib`), to distinguish them from the import libraries. These static libraries would only be used for the internal tests and not installed later on.

If building two library versions is undesired on this notoriously slow platform, then one could add a --static-tests option for windows, which is disabled by default.

**References**

This request originated from a discussion with @levitte in #7462, see in particular

* https://github.com/openssl/openssl/pull/7462#issuecomment-432021865
* https://github.com/openssl/openssl/pull/7462#issuecomment-432106224
* https://github.com/openssl/openssl/pull/7462#issuecomment-432118520

### PR DESCRIPTION
**Pull request by levitte**
When building shared libraries on Windows, we had a clash between
'libcrypto.lib' the static routine library and 'libcrypto.lib' the
import library.

We now change it so the static versions of our libraries get '_static'
appended to their names.  These will never get installed, but can
still be used for our internal purposes, such as internal tests.

When building non-shared, the renaming mechanism doesn't com into
play.  In that case, the static libraries 'libcrypto.lib' and
'libssl.lib' are installed, just as always.

Fixes #7492

##### Checklist

- [ ] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

**Review by t-j-h — APPROVED** [review]
Please correct "com" to "come" in the commit message before pushing

**Comment by levitte** [comment]
Will do

File: test/build.info [hunk 1]
```diff
@@ -436,10 +436,8 @@ INCLUDE_MAIN___test_libtestutil_OLB = /INCLUDE=main
   # available through the shared library (at least on Linux, Solaris, Windows
   # and VMS, where the exported symbols are those listed in util/*.num), these
   # programs are forcibly linked with the static libraries, where all symbols
-  # are always available.  This excludes linking these programs natively on
-  # Windows when building shared libraries, since the static libraries share
-  # names with the DLL import libraries.
-  IF[{- $disabled{shared} || $target{build_scheme}->[1] ne 'windows' -}]
+  # are always available.
+  IF[1]
```
Comment (mspncp) [inline test/build.info]: Do you intend to retain the possibily to revert the change? Or why didn't you remove the if block and reindent?

Comment (levitte) [reply test/build.info]: I chose for minimal change here.  I think there is value in grouping internals testers (you may have noticed the recipes with "internal" in their names too).

I'll leave a more thorough cleanup for another PR (and possibly someone else :wink:)

**Comment by levitte** [comment]
Merged.

master:
b3023ced6b6a4aece6f4d4ec1f6a93b1c03712b6 Windows: Produce a static version of the public libraries, always

1.1.1:
85299451cbdcb734e67adf14c64597a64dd11737 Windows: Produce a static version of the public libraries, always
